### DISCUSSION TITLE
Downgrade markdown version (TexLive)

### DISCUSSION THREAD

**Discussion by nopria** [Witiko/markdown#449] [Q&A] [answered]
I need to downgrade markdown version in TexLive because bug #447 prevents me totally from working. TexLive allows me to downgrade a package with

    tlmgr restore markdown

which shows available backups of the package, and

    tlmgr restore markdown <version number>

which actually restores the specified version, but my backups are not **old enough**. The markdown versions I'm looking for is about of mid 2023, those versions when you had to use option `inlineFootnotes=true` instead of `inlineNotes=true` as required by current version.

In my directory `C:\texlive\2023\tlpkg\backups\` I have a file `markdown.r71137.tar.lz4` which is a previous version but, as I said before, is not old enough. Is anybody able to provide from her/his system a similar backup file with an older version of markdown?

Thanks in advance for any help.

**Comment by Witiko** [comment, marked as answer]
I am sorry about your trouble. You can download historical versions of the Markdown package from [Releases](https://github.com/Witiko/markdown/releases). I suggest that you download the archive [`markdown.zip`](https://github.com/Witiko/markdown/releases/download/2.23.0/markdown.zip) with [version 2.23.0 from April 2023](https://github.com/Witiko/markdown/releases/tag/2.23.0). This version should predate [the introduction of CommonMark](https://github.com/Witiko/markdown/pull/226) in version 3.0.0-alpha, which is the likely cause behind https://github.com/Witiko/markdown/issues/447. You can install the older version as outlined in section "Installation" in the file `markdown.html` that is included in the archive.

**Comment by nopria** [comment]
Thanks, I use TexLive in Windows, as far as you know is there an alternative to command `make base` needed for manual installation?

**Reply by Witiko** [reply]
There is not. However, the step `make base` should be unnecessary, since it produces files that you can download from Releases, see the links in my answer.

**Reply by Witiko** [reply]
@nopria: In https://github.com/Witiko/markdown/pull/450, I extended the user manual to document that the Markdown package can be installed from GitHub releases:

![scrot](https://github.com/Witiko/markdown/assets/603082/c83e3a40-b852-44c0-b0bc-b5f8b8aa7d2d)

**Comment by nopria** [comment]
Thanks, just copying the specified files (taken directly from zipped release) to the document directory, as described in **Portable Installation**, works. Very easy and it does not affect current installation, allowing to restore updated release of package when bug #447 will be solved.
